File: golang/longest_harmonious_subsequence.go
package golang

func findLHS(nums []int) int {
	ans := 0
	d := make(map[int]int)

	for _, v := range nums {
		if _, ok := d[v]; ok {
			d[v]++
		} else {
			d[v] = 1
		}
	}

	for k := range d {
		if _, ok := d[k+1]; ok {
			ans = Max(ans, d[k]+d[k+1])
		}
	}

	return ans
}

File: python3/longest_harmonious_subsequence.py
from typing import List

"""
Space   : O(n)
Time    : O(n)
"""

class Solution:
    def findLHS(self, nums: List[int]) -> int:
        ans = 0
        d = {}
        
        for x in nums:
            if x not in d:
                d[x] = 1
            else:
                d[x] += 1
                
        for k in d.keys():
            if k+1 in d:
                ans = max(ans, d[k] + d[k+1])
        
        return ans
